SystemDocumentation/SystemDocs.txt

```diff
@@ -0,0 +1,206 @@
+
+addToManagerSearchQuery, this will do the same thing as above but for the manager only.
+
+main.py -
+Overview: This is the brain of our program. It handles all the internet requests, and handles all of the routes for the system.
+All of the program connects to the main.
+
+Use Cases: Its hard to gauge use cases for this function since none of its functions can really be called but we will still go over what they do.
+We will be covering what the routes do.
+
+@app.route("/managerPage")
+This route will take us to the manager page and will render that page and pass all necessary inputs in.
+
+@app.route("/searchInventoryById", methods=["POST"])
+This route will ask the database class for its method getById and grabs a product id passed and gives back the row of information
+containing that product id.
+
+@app.route("/searchInventoryByName", methods=["POST"])
+This route will ask the database class for its method getByName and grabs a name passed and gives back the row(s) of information
+containing that product name.
+
+@app.route("/searchEmailsByName", methods=["POST"])
+This route will ask the database class for its method getUser and grabs an email passed and gives back the row of information
+containing that user.
+
+@app.route("/login", methods=["GET","POST"])
+Uses the user login package to log a user in and create the session objects for that user.
+
+@app.route("/logout")
+This route will log the user out and clear their session objects
+
+@app.route("/pos")
+This route will render the pos.html template and gets passed all necessary inputs.
+
+@app.route("/cart", methods=["POST"])
+This method is a little more complicated, This method takes in input from the cashier, which is the product id. Looks that product up in the
+database, then adds it to the screen where the user and cashier can see their total amount. This also keeps track of what is in the
+users cart.
+
+
+@app.route("/checkout")
+This method leads the cashier from the cart to the checkout page. This route renders the checkout page
+
+@app.route("/methodOfPayment")
+Here we have the last method for the cashier to hit, this method will accept a form of payment for the items on the checkout page. After
+the cashier has entered payment it will clear the cart and the total for the current person.
+
+@app.route("/search")
+This method will take input from the user and search the database for what the user is looking by
+Possible search queries are:
+price
+product_id
+product name
+supplier
+It will return what the database finds in the table on the POS page
+
+
+@app.route("/profitReport")
+This method does pretty much what it says. It gives back a profit report of all the items in the inventory.
+
+@app.route("/managerSearch")
+This does the same thing as before, but it searches the database for the manager.
+
+@app.route("/createSale")
+This will create sales and lower or give price decreases on products in the database. Only the manager can access this functionality.
+
```

Comment:
I think we are missing the `/closeError`, `/orderMoreProducts`, `/transactions`, `/managerUpdate`, and `/cartClear` endpoints here.